packages/eslint-plugin/docs/rules/prefer-optional-chain.md

```diff
@@ -187,7 +187,7 @@ When this option is `true` the rule will check operands that are typed as `boole
 
 #### ❌ Incorrect for `checkBoolean: true`
 
-```ts option='{ "checkBoolean": true }'
+```ts option='{ "checkBoolean": true }' skipValidation
 declare const thing: boolean;
 
 thing && thing.toString();
```

Comment:
TODO: There are currently no lint errors for this code example ([playground](https://typescript-eslint.io/play/#eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5ujABtcNhy9o-aJAA0ylVhXZI0xGQDWAIX78kopvPxxEu7AF9nAXWVuXQA&code=CYUwxgNghgTiAEYD2A7AzgF3hgFgSxQHMAueAIySQhChQG4AoB3Aw%2BAMne3yIDoMkAZQwxWACgCUdIA)). If this is the desired behavior for boolean values, then we should change this example. Otherwise we should file an issue about this problem.

Reply:
https://github.com/typescript-eslint/typescript-eslint/blob/5e7ec8f23c511f5bdfc51c6479e8daa3526550e9/packages/eslint-plugin/src/rules/prefer-optional-chain-utils/gatherLogicalOperands.ts#L76-L86

https://github.com/typescript-eslint/typescript-eslint/pull/8497/commits/c28fdf2d6b6ef022ba30256d1edd0328df075108